src/python_testing/TC_CNET_4_1.py

```diff
@@ -0,0 +1,171 @@
+#
+#    Copyright (c) 2025 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import logging
+
+import chip.clusters as Clusters
+from chip.clusters.Types import NullValue
+from chip.testing import matter_asserts
+from chip.testing.matter_testing import MatterBaseTest, TestStep, default_matter_test_main, has_feature, run_if_endpoint_matches
+from mobly import asserts
+
+logger = logging.getLogger(__name__)
+
+
+class TC_CNET_4_1(MatterBaseTest):
+    """[TC-CNET-4.1] [Wi-Fi] Verification for attributes check [DUT-Server].
+
+    Example usage:
+        ```bash
+        python src/python_testing/TC_CNET_4_1.py --commissioning-method ble-wifi --qr-code <qrcode> \
+            --wifi-ssid <wifissid>  --wifi-passphrase <wifipasspgrase> --endpoint <endpointvalue>
+        ```
+        Where `<endpoint_value>` should be replaced with the actual endpoint
+        number for the Network Commissioning cluster on the DUT.
+    """
+
+    def desc_TC_CNET_4_1(self) -> str:
+        return "[TC-CNET-4.1] [Wi-Fi] Verification for attributes check [DUT-Server]"
+
+    def steps_TC_CNET_4_1(self) -> list[TestStep]:
+        steps = [
+            TestStep(1, "TH reads the MaxNetworks attribute from the DUT",
```

Comment:
Can you please add a test step in here that says to commission the DUT and has is_commissioning set to True? The TH folks are using that as a proxy to understand if the test requires the device to be commissioned.